### DISCUSSION TITLE
Images error - can't evaluate field MediaType in type string

### DISCUSSION THREAD

**Discussion by pensivedog** [gethyas/hyas#467] [Q&A]
After upgrading an existing Hyas site to use Hugo 0.123.7, I get the following error:

```
execute of template failed: template: partials/picture.html:233:19: executing "partials/picture.html" at <$l.MediaType>: can't evaluate field MediaType in type string
```
My images are in page bundles and the call in my single template looks like this:

```
{{ $image := .Page.Resources.GetMatch .Params.featured_image }}
{{ if $image }}
{{ partial "picture" (dict
            "page" .
            "src" $image
            "alt" .Title
            "class" "my-classes"
            "fetchpriority" "high"
            "loading" "eager"
            "sizes" "auto"
            )}}
{{ end }}
```

**Comment by h-enk** [comment]
Why define which image to show, if you can put `feature`, `cover`, or `thumbnail` in the filename — for example:

```html
{{- $images := .Resources.ByType "image" -}}
{{- $featured := $images.GetMatch "*feature*" -}}
{{- if not $featured }}{{ $featured = $images.GetMatch "{*cover*,*thumbnail*}" }}{{ end -}}
{{- if $featured -}}
  {{ partial "picture" (dict
    "page" .
    "src" $featured
    "alt" .Title
    "class" "my-classes"
    "fetchpriority" "high"
    "loading" "eager"
    "sizes" "auto"
  )}}
{{ end -}}
```

**Reply by pensivedog** [reply]
Yes, that's a better approach, but this is an existing website with thousands of user-uploaded images in page bundles. In the future we'll try to programmatically add one of these to the file name when images are added by users, but for now we'd like to get the existing setup working.

**Reply by h-enk** [reply]
I get it.

Your code works when `featured_image` is set in the frontmatter like so:

```md
---
featured_image: "my-featured-image.jpg"
---
```

However, I suspect this is how you have it set up:

```md
---
featured_image: ["my-featured-image.jpg"]
---
```

Then you would need to get `featured_image` like so — for example:

```html
{{ $image := .Resources.GetMatch (printf "**%s" (index .Params.featured_image 0)) }}
```

**Reply by pensivedog** [reply]
The front matter is set up so that `featured_image` is always a single image, not an array, as in:

```
---
featured_image: "my-featured-image.jpg"
---
```
When running `npm run dev` the complete error text is:
```
ERROR render of "page" failed: execute of template failed: template: articles/single.html:22:15: executing "main" at <partial "picture" (dict "page" . "src" $image "alt" .Title)>: error calling partial: "C:\mypath\node_modules\@hyas\images\layouts\partials\picture.html:233:19": execute of template failed: template: partials/picture.html:233:19: executing "partials/picture.html" at <$l.MediaType>: can't evaluate field MediaType in type string
Built in 917 ms
Error: error building site: render: failed to render pages: render of "page" failed: execute of template failed: template: articles/single.html:22:15: executing "main" at <partial "picture" (dict "page" . "src" $image "alt" .Title)>: error calling partial: "C:\mypath\node_modules\@hyas\images\layouts\partials\picture.html:233:19": execute of template failed: template: partials/picture.html:233:19: executing "partials/picture.html" at <$l.MediaType>: can't evaluate field MediaType in type string
```
Not sure what else to try...

**Reply by h-enk** [reply]
Can't reproduce your error/your code is working fine for me.

Repository: https://github.com/h-enk/doks-467/blob/main/layouts/_default/single.html#L33-L45
Deployed site: https://doks-467.netlify.app/docs/guides/example-guide/

Did you set the Image integration parameters correctly? https://github.com/h-enk/doks-467/blob/main/config/_default/params.toml#L103-L112

It really would help if you could share your repo

**Reply by pensivedog** [reply]
Thank you for the reply! It's a private repo but I was able to to solve it... the `lqip` parameter was missing. I added it and the error went away :)
